pkg/antctl/raw/check/installation/test_allegressdeny.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2024 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package installation
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"antrea.io/antrea/pkg/antctl/raw/check"
+)
+
+type AllEgressDenyConnectivityTest struct{}
+
+func init() {
+	RegisterTest("all-egress-deny-connectivity", &AllEgressDenyConnectivityTest{})
```

Comment:
I haven't reviewed the PR yet, but all tests should be independent and we should not mandate any specific execution order.

Reply:
Make sense, @kanha-gupta could you check and make sure there is no dependency between each test? In your current implementation, I think your NP tests depends on Pod connectivity tests deployment?

Reply:
Hi @luolanzone, Currently NP tests utilises Pod connectivity deployments that is client pod, echo same node pod and echo other node pod. Each NP test that is conducted first applies NP to required deployment and at the end, it deletes before proceeding to next test. This makes sure No error occurs and no test is executed with NP still applied to deployments.  This would also help when we expand the number of NP test cases.
Should we create new deployments for NP ? I believe it would follow the same format of 2 deployments on first node and 1 deployment on second node ?